src/pytest_plugins/filler/filler.py

```diff
@@ -384,12 +629,12 @@ def pytest_configure(config):
     # Initialize fixture output configuration
     config.fixture_output = FixtureOutput.from_config(config)
 
-    # Determine the filling phase
-    config.previous_filling_phases, config.current_filling_phase = _determine_filling_phases(
-        generate_pre_alloc_groups=config.getoption("generate_pre_alloc_groups"),
-        use_pre_alloc_groups=config.getoption("use_pre_alloc_groups"),
-        generate_all_formats=config.getoption("generate_all_formats"),
-    )
+    # Initialize global filling session
+    global _filling_session
+    _filling_session = FillingSession(config)
+
+    # Store session on config for access
+    config.filling_session = _filling_session
```

Comment:
This seems redundant. Do we need to have a global AND store it in pytest config.

Reply:
I think we should remove the global and have all hooks use the object from pytest.config. This is more consistent with how we handle such global objects now.